org.eclipse.xtext.testing/src/org/eclipse/xtext/testing/extensions/InjectionExtension.java

```diff
@@ -49,66 +52,85 @@
 	 * </p>
 	 */
 	private static ClassToInstanceMap<IInjectorProvider> injectorProviderClassCache = MutableClassToInstanceMap.create();
+	
+	/**
+	 * @since 2.24
+	 */
+	protected static class RegistryReset implements CloseableResource {
+		protected IRegistryConfigurator resetter;
+
+		public RegistryReset(IRegistryConfigurator resetter) {
+			this.resetter = resetter;
+			resetter.setupRegistry();
+		}
+
+		@Override
+		public void close() throws Throwable {
+			if (resetter != null) {
+				resetter.restoreRegistry();
+				resetter = null;
+			}
+		}
+	}
 
+	/**
+	 * @since 2.24
+	 */
 	@Override
-	public void beforeEach(ExtensionContext context) throws Exception {
+	public void postProcessTestInstance(Object testInstance, ExtensionContext context) throws Exception {
 		IInjectorProvider injectorProvider = getOrCreateInjectorProvider(context);
-		if (injectorProvider instanceof IRegistryConfigurator) {
-			final IRegistryConfigurator registryConfigurator = (IRegistryConfigurator) injectorProvider;
-			registryConfigurator.setupRegistry();
-		}
 		if (injectorProvider != null) {
+			setupRegistry(injectorProvider, context);
 			Injector injector = injectorProvider.getInjector();
 			if (injector != null) {
-				Object testInstance = context.getRequiredTestInstance();
 				injector.injectMembers(testInstance);
-				try {
-					TestInstances requiredTestInstances = context.getRequiredTestInstances();
-					for (Object o : requiredTestInstances.getEnclosingInstances()) {
-						injector.injectMembers(o);
-					}
-				} catch (NoSuchMethodError e) {
-					if (!Modifier.isStatic(testInstance.getClass().getModifiers())) {
-						if (testInstance.getClass().getDeclaringClass() != null) {
-							throw new ExtensionConfigurationException("Injection of nested classes needs Junit5 >= 5.4", e);
-						}
-					}
-					// OK, getRequiredTestInstances is not there in Junit5 < 5.4
-				}
```

Comment:
is this no longer an issue?
we dont package at umbrella

Reply:
Should be fine since we only use 5.0 APIs now.